manager/backend/api.py

```diff
@@ -1,65 +1,175 @@
+						token=token,
+						displayName=request.data["displayName"],
+						github=request.data["github"],
+						host = request.META['HTTP_HOST'],
+						)
+
+		# Save the author information into the database
+		author.url = "http://"+str(author.host)+"/"+str(author.id)
+		author.save()
+		# Serialize the author data for a POST response
+		authorData = AuthorSerializer(author, context=self.get_serializer_context()).data
+
+		return Response(authorData)
+
+class UserAPI(generics.RetrieveAPIView):
+
+	permission_classes = [
+		permissions.IsAuthenticated,
+	]
+
+	serializer_class = UserSerializer
+
+	def get_object(self):
+		return self.request.user
+
+
+
+
 class AuthorViewSet(viewsets.ModelViewSet):
-    
-    queryset = Author.objects.all()
+	"""
+	This class specifies the view for the Post objects. This will run methods to retrieve and edit DB rows and return correctly formatted HTTP responses
+	"""
+	# Specifies the query set on which this view can act
+	queryset = Author.objects.all()
+
+	# Specifies the permissions needed to access and modify the data
+	permission_classes = [
+		permissions.IsAuthenticated
+	]
+
+	# Specifies the serializer used to return the correctly formatted JSON response body
+	serializer_class = AuthorSerializer
+
+	# Specifies the lookup field to use the in the database
+	lookup_field = 'id'
+
+	def update(self, request, *args, **kwargs):
+		"""
+		This method will be called when a POST request is received for a specific author.
+		"""
+
+		# Code to extract specifically the author's id value from the url in the JSON request's id field
+		author_id = request.data["id"]
+		newID = author_id.split("/")[-2]
+
+		# Update the request object's data so the id field is correctly formatted to be found in the DB
+		request.data.update({"id": newID})
+
+		return super().update(request, *args, **kwargs)
+
+
+# Get Author API
+class LoginAPI(viewsets.ModelViewSet):
+
+	queryset = Author.objects.all()
+
+	permission_classes = [
+		#permissions.AllowAny
+		permissions.AllowAny
+	]
+
+	serializer_class = AuthorSerializer
 
-    permission_classes = [
-        permissions.AllowAny
-    ]
-    serializer_class = AuthorSerializer
 
-    lookup_field = 'id'
 
-# Register API
-class RegisterAPI(generics.GenericAPIView):
-  serializer_class = RegisterSerializer
-
-  def post(self, request, *args, **kwargs):
-    serializer = self.get_serializer(data=request.data)
-    serializer.is_valid(raise_exception=True)
-    user = serializer.save()
-    token = Token.objects.create(user=user)
-    userData = UserSerializer(user, context=self.get_serializer_context()).data
-    return Response({
-      "user": userData
-    })
 class PostViewSet(viewsets.ModelViewSet):
+	"""
+	This class specifies the view for the Post objects. This will run methods to retrieve and edit DB rows and return correctly formatted HTTP responses
+	"""
+
+	# Specifies the permissions required to access the data
+	permission_classes = [
+		permissions.AllowAny
```

Comment:
this is just temporary right?

Reply:
This is temporary until we know how we're going to authenticate friends/followers on our server and other servers.